### PR TITLE
Include draft orders in voucher usage

### PR DESCRIPTION
**Pull request by IKarbowiak**
Feature branch for including draft orders in voucher calculations.

RFC: https://github.com/saleor/saleor/issues/13713

Docs: https://github.com/saleor/saleor-docs/pull/1002

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases

### CODE REVIEW

File: saleor/graphql/channel/mutations/channel_update.py [hunk 1]
```diff
@@ -218,9 +222,26 @@ def _update_warehouses(cls, instance, cleaned_data):
...
+        - True -> False: decrease voucher usage of all vouchers associated with
+        draft orders.
+        - False -> True: disconnect vouchers from all draft orders.
+        """
+        current_flag = cleaned_input.get("include_draft_order_in_voucher_usage")
+        previous_flag = cleaned_input.get("prev_include_draft_order_in_voucher_usage")
+        if current_flag is False and previous_flag is True:
+            decrease_voucher_code_usage_of_draft_orders(instance.id)
+        elif current_flag is True and previous_flag is False:
+            disconnect_voucher_codes_from_draft_orders(instance.id)
```
Comment (korycins) [inline saleor/graphql/channel/mutations/channel_update.py:239]: Are we sure about this? 
When I disable `include_draft_order_in_voucher_usage`, I would expect the voucher.usage_limit will not increase when using it with draft order. But currently, it fully removes the vouchers from the draft orders, and it sounds to me like a separate setting or enum setting.

Comment (korycins) [reply saleor/graphql/channel/mutations/channel_update.py:239]: Nevermind. I understand the flow now.

Comment (IKarbowiak) [reply saleor/graphql/channel/mutations/channel_update.py:239]: When you disabling this setting, the voucher usage is decreased, as draft order's vouchers are not counted into voucher usage anymore.

When switching on this setting, all vouchers are removed from draft orders, as it to many cases to handle. For example when the voucher will have `usage_limit=2` and 3 draft orders will have this voucher assigned there is no easy wat to handle such case.

File: saleor/graphql/channel/mutations/channel_update.py [hunk 2]
```diff
@@ -218,9 +222,26 @@ def _update_warehouses(cls, instance, cleaned_data):
         if remove_warehouses:
             instance.warehouses.remove(*remove_warehouses)
 
+    @classmethod
+    def _update_voucher_usage(cls, cleaned_input, instance):
+        """Update voucher code usage.
+
+        When the 'include_draft_order_in_voucher_usage' flag is changed:
```
[the remark this answers is not in the document]
Comment (IKarbowiak) [reply saleor/graphql/channel/mutations/channel_update.py:229]: Right, should be included

[the remark this answers is not in the document]
Comment (IKarbowiak) [reply saleor/graphql/channel/mutations/channel_update.py:229]: Done

File: saleor/graphql/channel/mutations/utils.py [hunk 3]
```diff
@@ -40,24 +41,26 @@ def clean_delete_expired_orders_after(delete_expired_orders_after: int) -> timed
...
+    channel_settings = [
+        "automatically_confirm_all_new_orders",
+        "automatically_fulfill_non_shippable_gift_card",
+        "mark_as_paid_strategy",
+        "default_transaction_flow_strategy",
+        "allow_unpaid_orders",
+        "include_draft_order_in_voucher_usage",
+    ]
+
+    for field in channel_settings:
```
[the remark this answers is not in the document]
Comment (IKarbowiak) [reply saleor/graphql/channel/mutations/utils.py:55]: Done

**Review by korycins — APPROVED** [review]

**Review by kadewu — APPROVED** [review]

**Review by yellowee — APPROVED** [review]

**Pull request merged**